applications/OptimizationApplication/custom_utilities/container_variable_data_holder/container_variable_data_holder.cpp

```diff
@@ -0,0 +1,657 @@
+        << "      Right operand data: " << rOther << "\n";
+
+    ContainerVariableDataHolder<TContainerType, TContainerDataIO> result(*(this->mpModelPart));
+    result.mDataDimension = this->mDataDimension;
+    result.mData.resize(this->mData.size(), false);
+    IndexPartition<IndexType>(this->GetContainer().size()).for_each([&](const IndexType Index) {
+        for (IndexType i = 0; i < this->mDataDimension; ++i) {
+            double& v1 = result.mData[Index * this->mDataDimension + i];
+            const double v2 = this->mData[Index * this->mDataDimension + i];
+            for (IndexType j = 0; j < rOther.mDataDimension; ++j) {
+                v1 = v2 / rOther.mData[Index * rOther.mDataDimension + j];
+            }
+        }
+    });
+
+    return result;
+}
+
+template <class TContainerType, class TContainerDataIO>
+ContainerVariableDataHolder<TContainerType, TContainerDataIO>& ContainerVariableDataHolder<TContainerType, TContainerDataIO>::operator/=(const ContainerVariableDataHolder<TContainerType, TContainerDataIO>& rOther)
+{
+    KRATOS_ERROR_IF(&this->GetModelPart() != &rOther.GetModelPart())
+        << "Mismatching model parts found in substraction.\n"
+        << "      Left operand data : " << *this << "\n"
+        << "      Right operand data: " << rOther << "\n";
+
+    KRATOS_ERROR_IF(this->mDataDimension != rOther.mDataDimension && (this->mDataDimension * rOther.mDataDimension == 0 || rOther.mDataDimension != 1))
+        << "Dimension mismatch in operands for /. Right operand data dimension should be 1 or equal to left operand dimension.\n"
+        << "      Left operand data : " << *this << "\n"
+        << "      Right operand data: " << rOther << "\n";
+
+    IndexPartition<IndexType>(this->GetContainer().size()).for_each([&](const IndexType Index) {
+        for (IndexType i = 0; i < this->mDataDimension; ++i) {
+            double& v1 = this->mData[Index * this->mDataDimension + i];
+            for (IndexType j = 0; j < rOther.mDataDimension; ++j) {
+                v1 /= rOther.mData[Index * rOther.mDataDimension + j];
+            }
+        }
+    });
+
+    return *this;
+}
+
+template <class TContainerType, class TContainerDataIO>
+ContainerVariableDataHolder<TContainerType, TContainerDataIO> ContainerVariableDataHolder<TContainerType, TContainerDataIO>::operator/(const double Value) const
+{
+    KRATOS_ERROR_IF(std::abs(Value) < std::numeric_limits<double>::epsilon())
+        << "Division by zero.\n"
+        << "      Left operand data : " << *this << "\n"
+        << "      Divisor           : " << Value << "\n";
+
+    ContainerVariableDataHolder<TContainerType, TContainerDataIO> result(*(this->mpModelPart));
+    result.mDataDimension = this->mDataDimension;
+    result.mData.resize(this->mData.size(), false);
+    IndexPartition<IndexType>(this->mData.size()).for_each([&](const IndexType Index) {
+        result.mData[Index] = this->mData[Index] / Value;
```

Comment:
I'll reiterate that I don't think you should implement these operations yourself.

As an extra remark, division is several times slower than multiplication, and the compiler won't help you here because `Value` is not a literal. I'd compute the inverse of `Value` outside the loop, and multiply with that inverse inside.

Reply:
done :)